addon/models/user.js

```diff
@@ -1,202 +1,263 @@
+  @use _administeredCurriculumInventoryReports = new ResolveAsyncValue(() => [
+    this.administeredCurriculumInventoryReports,
+  ]);
+  @use _directedSchools = new ResolveAsyncValue(() => [this.directedSchools]);
+  @use _administeredSchools = new ResolveAsyncValue(() => [this.administeredSchools]);
 
   /**
    * Checks if a user is linked to any non-student things
-   * @property performsNonLearnerFunction
-   * @type {Ember.computed}
-   * @public
    */
-  performsNonLearnerFunction: computed(
-    'directedCourses.[]',
-    'administeredCourses.[]',
-    'administeredSessions.[]',
-    'instructedLearnerGroups.[]',
-    'instructorGroups.[]',
-    'instructedOfferings.[]',
-    'directedPrograms.[]',
-    'programYears.[]',
-    'administeredCurriculumInventoryReports.[]',
-    'directedSchools.[]',
-    'administeredSchools.[]',
-    function () {
-      const directedCourses = this.hasMany('directedCourses').ids();
-      const administeredCourses = this.hasMany('administeredCourses').ids();
-      const administeredSessions = this.hasMany('administeredSessions').ids();
-      const instructedLearnerGroups = this.hasMany('instructedLearnerGroups').ids();
-      const instructorGroups = this.hasMany('instructorGroups').ids();
-      const instructedOfferings = this.hasMany('instructedOfferings').ids();
-      const directedPrograms = this.hasMany('directedPrograms').ids();
-      const programYears = this.hasMany('programYears').ids();
-      const administeredCurriculumInventoryReports = this.hasMany(
-        'administeredCurriculumInventoryReports'
-      ).ids();
-      const directedSchools = this.hasMany('directedSchools').ids();
-      const administeredSchools = this.hasMany('administeredSchools').ids();
-
-      return (
-        !isEmpty(directedCourses) ||
-        !isEmpty(administeredCourses) ||
-        !isEmpty(administeredSessions) ||
-        !isEmpty(instructedLearnerGroups) ||
-        !isEmpty(instructorGroups) ||
-        !isEmpty(instructedOfferings) ||
-        !isEmpty(directedPrograms) ||
-        !isEmpty(programYears) ||
-        !isEmpty(administeredCurriculumInventoryReports) ||
-        !isEmpty(directedSchools) ||
-        !isEmpty(administeredSchools)
-      );
-    }
-  ),
-
-  fullName: computed('fullNameFromFirstMiddleInitialLastName', 'displayName', function () {
-    return this.displayName ? this.displayName : this.fullNameFromFirstMiddleInitialLastName;
-  }),
-
-  fullNameFromFirstMiddleInitialLastName: computed(
-    'firstName',
-    'middleName',
-    'lastName',
-    function () {
-      if (!this.firstName || !this.lastName) {
-        return '';
+  get performsNonLearnerFunction() {
+    deprecate(
+      `user.performsNonLearnerFunction called, should be read from current-user session.`,
+      false,
+      {
+        id: 'common.user-performs-non-learner-function',
+        for: 'ilios-common',
+        until: '65',
+        since: '62.0.0',
       }
+    );
+    return Boolean(
```

Comment:
i don't think this needs to be wrapped in a Boolean constructor. the result of the disjunction is always a boolean.